src/artemis/document.cljc

```diff
@@ -79,18 +79,20 @@
   [doc mapping]
   (vary-meta doc assoc ::operation-mapping mapping))
 
-(defn merge-selection-sets [s1 s2]
-  (assoc s1 :selection-set
-            (->> (concat (:selection-set s1) (:selection-set s2))
-                 set
-                 (into []))))
+(defn combine-selections [s1 s2]
```

Comment:
This might be able to be more succinctly written as:
```
(merge-with #(-> (concat %1 %2) set (into [])) s1 s2)
```

Reply:
nice

Reply:
this doesn't work because it'll also try to concat regular keyword/string values (such as `:field` below) and fail 
```
{:node-type :field,
 :field-name "numUnreadMessages"}
```